carbon-design-system/carbon
# fix: added floating UI to menu and menuButton

Closes #15521
Closes #14296  

This PR enhances MenuButton experimental autoAlign functionality to automatically position the floating element to be within view, even on scroll.

This uses [floating-ui](https://floating-ui.com/) and relies solely on the fixed strategy, using position: fixed on the floating element. In the majority of cases, MenuButton floating elements will no longer be clipped by parents with overflow: hidden.

Changed

- Use floating-ui for positioning  floating element.
- Feed floating-ui the values of menuButton custom properties (flip,size) when possible.
- Added a prop named legacyAutoalign to menu that specifies if the parent element is using floating Ui for positioning the -menu or it sending X,Y coordinates. On the basis of it calculation of positioning the menu is  made conditional.

Removed

- Sending X and Y coordinates for the floating element (Menu) from MenuButton. All the calculations like width of floating element , X and Y positions are calculated by floating UI itself.

Testing / Reviewing

- View the menuButton stories, there is autoalign experiment story in there. Ensure it works as intended and there's no regressions.
- Make sure that the menu does not detach itself from menuButton.
